components/formats-bsd/test/loci/formats/utests/FileStitcherTest.java

```diff
@@ -193,4 +212,43 @@ public void testStitch(Integer[] dims) throws IOException, FormatException {
     check(pattern, filenames.toArray(new String[filenames.size()]), dims);
   }
 
+  @Test
+  public void testUnderlyingReaders() throws IOException, FormatException {
+    FakeReader reader = new FakeReader();
+    FileStitcher fs = new FileStitcher(reader);
+    assertNotNull(fs.getUnderlyingReaders());
+    fs.setId("test_z<0-2>.fake");
+    assertNotNull(fs.getUnderlyingReaders());
+    fs.close();
+  }
+
+  @Test
+  public void testOptionsExplicit() throws IOException, FormatException {
+    DynamicMetadataOptions opt = new DynamicMetadataOptions();
+    opt.set(KEY, VALUE);
+    FileStitcher fs = new FileStitcher();
+    fs.setMetadataOptions(opt);
+    fs.setId("test_z<0-2>.fake");
+    for (IFormatReader r: fs.getUnderlyingReaders()) {
+      MetadataOptions rOpt = r.getMetadataOptions();
+      assertTrue(rOpt instanceof DynamicMetadataOptions);
+      String v = ((DynamicMetadataOptions) rOpt).get(KEY);
+      assertNotNull(v);
+      assertEquals(v, VALUE);
+    }
+    fs.close();
+  }
+
+  @Test(dataProvider = "levels")
+  public void testOptionsImplicit(MetadataLevel level)
+      throws IOException, FormatException {
+    FileStitcher fs = new FileStitcher();
+    fs.getMetadataOptions().setMetadataLevel(level);
+    fs.setId("test_z<0-2>.fake");
+    for (IFormatReader r: fs.getUnderlyingReaders()) {
+      assertEquals(r.getMetadataOptions().getMetadataLevel(), level);
+    }
+    fs.close();
+  }
```

Comment:
How should the options be handled in the case of closing/reopening? Does it fundamentally fall back to whether options have been set explicitly/implicitly before the second call to `setId` or are the options from the initial `setId()` call still used?